core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -481,28 +481,33 @@ class AutoTuner(
   private def calcOverallMemory(
       execHeapCalculator: () => Long,
       numExecutorCores: Int,
-      containerMemCalculator: () => Double): Either[String, (Long, Long, Long, Boolean)] = {
+      totalMemForExecExpr: () => Double): Either[String, (Long, Long, Long, Boolean)] = {
     // Set executor heap to be at least 2GB/core
     val executorHeap = Math.max(execHeapCalculator(),
       autoTunerConfigsProvider.DEF_HEAP_PER_CORE_MB * numExecutorCores)
-    val containerMem = containerMemCalculator.apply()
-    val containerMemLeftOverOffHeap = containerMem - executorHeap
+    // Our CSP instance map stores full node memory, but container managers
+    // (e.g., YARN) may reserve a portion. Adjust to get the memory
+    // actually available to the executor.
+    val actualMemForExec = {
+      totalMemForExecExpr.apply() * platform.fractionOfAvailableMemory
+    }.toLong
+    val offHeapMemSetByUser = platform.getTotalOffHeapMemoryMB(getAllProperties)
+    val execMemLeft = actualMemForExec - executorHeap - offHeapMemSetByUser
     var setMaxBytesInFlight = false
     // reserve 10% of heap as memory overhead
     var executorMemOverhead = (
-      executorHeap * autoTunerConfigsProvider.DEF_HEAP_OVERHEAD_FRACTION +
-        autoTunerConfigsProvider.DEF_PAGEABLE_POOL_MB
+      executorHeap * autoTunerConfigsProvider.DEF_HEAP_OVERHEAD_FRACTION
```

Comment:
Removing pageable pool as the comment said it is not used anymore and we are now handling offHeap explicitly.

cc: @tgravescs